dataactvalidator/scripts/load_cfda.py

```diff
@@ -1,54 +1,89 @@
 from dataactcore.logging import configure_logging
+import logging
 
 import boto3
+from botocore.handlers import disable_signing
+from botocore.exceptions import ClientError
+
+s3 = boto3.client('s3', )
+logger = logging.getLogger(__name__)
+
+CFDA_FILE_FORMAT = os.path.join('Assistance Listings', 'usaspendinggov', '%Y', '%m-%b',
+                                'AssistanceListings_USASpendingGov_PUBLIC_WEEKLY_%Y%m%d.csv')
+WEEKDAY_UPLOADED = 5  # datetime.weekday()'s integer representing the day it's usually uploaded (Saturday)
+DAYS_TO_SEARCH = 4 * 7  # 4 weeks
+LOCAL_CFDA_FILE = os.path.join('dataactvalidator', 'config', 'cfda_program.csv')
+LOCAL_CFDA_FILE2 = os.path.join('dataactvalidator', 'config', 'cfda_program2.csv')
+
+
+def find_latest_file(bucket, days_to_search=DAYS_TO_SEARCH):
+    # TODO: If/When the bucket is public, simply use the folder structure to find the latest file instead of guessing
+    # Check for the latest Saturday upload, otherwise manually look it up
+    today = datetime.today()
+    if today.weekday() == WEEKDAY_UPLOADED:
+        logger.info('Checking today\'s entry')
+        latest_file = today.strftime(CFDA_FILE_FORMAT)
+        if file_exists(bucket, latest_file):
+            return latest_file
+
+    logger.info('Checking last week\'s entry')
+    last_week = today - timedelta(7 - abs(today.weekday() - WEEKDAY_UPLOADED))
+    latest_file = last_week.strftime(CFDA_FILE_FORMAT)
+    if file_exists(bucket, latest_file):
+        return latest_file
+    else:
+        logger.info('Looking within the past {} days'.format(days_to_search))
+        try_date = today
+        while days_to_search > 0:
+            latest_file = try_date.strftime(CFDA_FILE_FORMAT)
+            if not file_exists(bucket, latest_file):
+                try_date = try_date - timedelta(1)
+                days_to_search -= 1
+            else:
+                break
+        if days_to_search == 0:
+            logger.error('Could not find cfda file within the past {} days.'.format(days_to_search))
+            return None
+        return latest_file
+
+
+def file_exists(bucket, src):
+    try:
+        bucket.Object(src).load()
+        return True
+    except ClientError:
+        return False
 
 
 def load_cfda():
-    # connect to host, default port
-    ftp = FTP('ftp.cfda.gov')
-    # anonymous FTP, the archive site allow general access
-    # user anonymous, password anonymous
-    ftp.login()
-
-    # change the directory to /usaspending/
-    ftp.cwd('usaspending')
-
-    data = []
-
-    # output the directory contents
-    ftp.dir('-t', data.append)
-
-    # get the most recent updated cfda file
-    file_listing = data[0]
+    gsa_connection = boto3.resource('s3', region_name='us-east-1')
+    # disregard aws credentials for public file
+    gsa_connection.meta.client.meta.events.register('choose-signer.s3.*', disable_signing)
+    gsa_bucket = gsa_connection.Bucket('falextracts')
```

Comment:
Should this live in the settings file? we may want to consider external settings/config files similar to broker so it won't require a code change/deploy to update external references.

Reply:
Makes sense. I can work with Ops to figure out the best way of adding these setting.